pkg/controller/compliance/compliance_controller.go

```diff
@@ -274,13 +280,30 @@ func (r *ReconcileCompliance) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Fetch the Authentication spec. If present, we use it to configure dex as an authentication proxy.
+	authentication, err := utils.GetAuthentication(ctx, r.client)
```

Comment:
In downstream controllers that rely on dex being properly configured, I believe we should be blocking them based on the Authentication.Status.Ready field that we set only after Dex is properly configured.